src/Timecode.php

```diff
@@ -0,0 +1,413 @@
+        }, $times);
+
+        $roundFrameRate = round($frameRate);
+        $frameCount = self::calculateFrameCount($roundFrameRate, $hours, $minutes, $seconds, $frames);
+
+        if ($dropFrame) {
+            $totalMinutes = (60 * $hours) + $minutes;
+
+            return (int) ($frameCount - (2 * ($totalMinutes - floor($totalMinutes / 10))));
+        }
+
+        return (int) $frameCount;
+    }
+
+    /**
+     * Create a new Timecode of a given seconds
+     *
+     * @param int $seconds
+     * @param float $frameRate
+     * @param bool $dropFrame
+     * @return self
+     */
+    public static function fromSeconds($seconds, $frameRate = null) : self
+    {
+        $frameRate = is_null($frameRate) ? self::$defaultFrameRate : $frameRate;
+
+        if (! is_numeric($seconds)) {
+            throw new \InvalidArgumentException('First argument must be a number');
+        }
+
+        return new self(intval($seconds * $frameRate), $frameRate);
+    }
+
+    /**
+     * Calculate frame count
+     *
+     * @param float $frameRate
+     * @param int $hours
+     * @param int $minutes
+     * @param int $seconds
+     * @param int $frames
+     * @return float
+     */
+    public static function calculateFrameCount(
+        float $frameRate,
+        int $hours,
+        int $minutes,
+        int $seconds,
+        int $frames
+    ) : float {
+        return ($frameRate * 60 * 60 * $hours)
+            + ($frameRate * 60 * $minutes)
+            + ($frameRate * $seconds)
+            + $frames;
+    }
+
+    /**
+     * @param float $frameRate
+     */
+    public static function setDefaultFrameRate(float $frameRate)
+    {
+        self::$defaultFrameRate = $frameRate;
+    }
+
+    /**
+     * @param float $frameRate
+     */
+    public static function setDefaultDropFrame(bool $dropFrame)
+    {
+        self::$defaultDropFrame = $dropFrame;
+    }
+
+    /**
+     * Convert the Timecode to its string representation.
+     *
+     * @return string
+     */
+    public function toString() : string
+    {
+        $dropFrameSymbol = $this->dropFrame ? ';' : ':';
+
+        return sprintf("%02d:%02d:%02d{$dropFrameSymbol}%02d", $this->hours, $this->minutes, $this->seconds, $this->frames);
+    }
+
+    /**
+     * Create a frame count based on your time type
+     *
+     * @param int|string|DateTime $time
+     * @return self
+     */
+    public function generateFrameCount($time) : self
+    {
+        $frameCount = null;
+
+        if (is_integer($time)) {
+            $frameCount = (int) $time;
+        } elseif ($time instanceof \DateTime) {
+            $midnight = clone $time;
+            $midnight->setTime(0, 0);
+
+            $frameCount = (int) (($time->getTimestamp() - $midnight->getTimestamp()) * $this->frameRate);
+        } elseif (is_string($time)) {
+            $frameCount = (int) self::frameCountFromTimecode($time, $this->frameRate, $this->dropFrame);
+        }
+
+        if ($frameCount === null) {
```

Comment:
Let's choose the best way to check `null` value:

- `null === $var` (Yoda condition)
- `is_null($var)`

I prefer `Yoda condition` (it's faster than `is_null`), what do you guys prefer?